### DISCUSSION TITLE
Refresh View page after Action

### DISCUSSION THREAD

**Discussion by donnyxray** [filamentphp/filament#6899] [General]
Currently I add this line to Actions on View pages:

```->after(fn() => $this->fillForm())```

Without it, the values on the view page are not updated after running the Action.

Is this the best way to do this? I feel like this is such a common use-case that there should be a less repetitive approach. Ideally, it's automatic and consistent with the List view. (I understand the Edit view is a bit different due to possible unsaved values).

**Comment by pboivin** [comment]
Maybe `refreshFormData()` 
https://filamentphp.com/docs/2.x/admin/pages/actions#refreshing-form-data

**Comment by donnyxray** [comment]
I had that at first, but it requires more code than my current solution.

I guess the reason it bugs me is because it's fully automatic in the List view. From my perspective the View page is a detailed look at a List item but should otherwise function the same. The Filament perspective is probably that View is a read-only version of the Edit page.

**Comment by AbdullahMostafa33** [comment]
add this in action
->after(fn($record, $livewire) => $livewire->data = $record->toArray()),
